app/auth/login/page.jsx

```diff
@@ -1,77 +1,53 @@
-"use client";
-import React, { useState } from "react";
-import { Box, Button, TextField } from "@mui/material";
-import styles from './page.module.css'
-import Layout from "../layout";
-import Link from "next/link";
+'use client'
+import styles from '@/app/auth/login/page.module.css'
+import { isPasswordValid, isUserValid } from '@/app/auth/utils/utils'
+import useActions from '@/app/hooks/useActions'
+import useShallowEqualSelector from '@/app/hooks/useShallowEqualSelector'
+import { userActions } from '@/app/redux/reducers/user'
+import * as auth from '@/app/rest_client/authClient'
+import { Box, Button, TextField } from '@mui/material'
+import Link from 'next/link'
 import { useRouter } from 'next/navigation'
+import { useEffect, useState } from 'react'
 
-export default function Login(props) {
-
+export default function Login() {
   const [userName, setUserName] = useState('')
   const [password, setPassword] = useState('')
 
-
+  const { login, loginFailure } = useActions(userActions)
   const router = useRouter()
+  const { isLogin } = useShallowEqualSelector((state) => state.user)
 
-
-  // Ham check cac chu viet hoa tra ve true
-  const containsUppercase = (str) => {
-    return /[A-Z]/.test(str);
-  }
-
-
-  // Ham check ky tu dac biet
-  const checkSpecialSymbol = (str) => {
-    return /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/.test(str)
-  }
-
-
-  // ham check space
-  const checkSpace = (str) => {
-    return /[ ]/.test(str);
-  }
-
-
-  // ham check validation cua username
-  const userValidation = () => {
-    if (containsUppercase(userName) || checkSpecialSymbol(userName) || userName.length === 0) {
-      return true
+  useEffect(() => {
+    if (isLogin) {
+      router.push('/home')
     }
-    if (!containsUppercase(userName) && !checkSpecialSymbol(userName) || !userName.length === 0) {
-      return false
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isLogin])
+
+  const submitLogin = async () => {
+    if (!isLogin) {
+      try {
+        const res = await auth.login({ userName, password })
+        login(res.data)
+      } catch (error) {
+        loginFailure(error.message)
+      }
     }
   }
 
-
-  //ham check validation cua password
-  function passValida() {
-    if (password.length === 0 || checkSpace(password)) {
-      return true;
-    }
-    else {
-      return false;
-    }
-  }
-
-
-  //ham handle click login button
-  const submitLogin = (e) => {
-    e.preventDefault()
-    router.push('/auth/FogotPassword')
-  }
-
-
   return (
-      <Box sx={{
+    <Box
+      sx={{
         height: '100vh',
```

Comment:
here, I would rather use minHeight: '100vh'
In case there is much content on screen, there will be no scrolling when you use height